src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -101,15 +109,27 @@ void BaseCouplingScheme::sendData(const m2n::PtrM2N &m2n, const DataMap &sendDat
   PRECICE_DEBUG("Number of sent data sets = {}", sentDataIDs.size());
 }
 
-void BaseCouplingScheme::receiveData(const m2n::PtrM2N &m2n, const DataMap &receiveData)
+void BaseCouplingScheme::receiveData(const m2n::PtrM2N &m2n, const DataMap &receiveData, bool recvInitialData)
 {
   PRECICE_TRACE();
   std::vector<int> receivedDataIDs;
   PRECICE_ASSERT(m2n.get());
   PRECICE_ASSERT(m2n->isConnected());
   for (const DataMap::value_type &pair : receiveData) {
     // Data is only received on ranks with size>0, which is checked in the derived class implementation
-    m2n->receive(pair.second->values(), pair.second->getMeshID(), pair.second->getDimensions());
+    auto recvBuffer = Eigen::VectorXd(pair.second->getSize());
+    m2n->receive(recvBuffer, pair.second->getMeshID(), pair.second->getDimensions());
+
+    // will be changed via https://github.com/precice/precice/pull/1414
+    if (recvInitialData) {
+      pair.second->storeValuesAtTime(time::Storage::WINDOW_START, recvBuffer);
+      pair.second->storeValuesAtTime(time::Storage::WINDOW_END, recvBuffer);
+    } else {
+      pair.second->clearTimeStepsStorage();
+      pair.second->storeValuesAtTime(time::Storage::WINDOW_END, recvBuffer);
+    }
+
+    pair.second->values() = recvBuffer; // @todo should happen somewhere else!
```

Comment:
Not 100% sure, but generally it would be nice to take care of putting the "right" data into `pair.second->values()` just before we move from the `cplscheme` package to another package (acceleration, mapping ...). That's the reason I marked it here. But it is also fine the way it is right now. It might become easier to put it somewhere else when we continue working on #1414 

See also aa2765b47c6ef81039ca41647f30814884e5ee34